design/export-functionality.md

```diff
@@ -0,0 +1,114 @@
+
+### Using the java client
+Reimplement the spark client export functionality by creating a new Exporter class that doesn't use spark.
+This class can be part of the java cliet, or a new "Export" client that uses the java client.
+
+The usage will be with java code (in a similar way of using custom code with the spark client).
+
+In order to use Java code, users will need to download it if necessary, learn it, and have it in their toolset.
+
+Usage example:
+
+```java
+class Export{
+    public static void main(String[]args){
+        Exporter exporter = new Exporter(apiClient, repoName, rootLocation);
+        exporter.exportAllFromBranch(branchName);
+        exporter.exportFrom(branchName, commitRef);
+    }
+}
+```
+
+Pros:
+- An intermediate solution that is compatible with the issue.
+
+Cons:
+- Require checking how spark is being used in the current implementation (in order to rewrite its functions). Not necessarily possible in a naive way.
+- Require developing and maintaining a second export functionality.
+
+### Using Rclone
+[Rclone](https://rclone.org/) is a command line program to sync files and directories between cloud providers.
+Users can gain the export functionality by using Rclone's [copy](https://rclone.org/commands/rclone_copy/), [sync](https://rclone.org/commands/rclone_sync/), and [check](https://rclone.org/commands/rclone_check/) commands.
+
+The copy command can be used to copy files from lakeFS to a designated object store location.
+The sync command can be used to export only the diff between a specific branch and a commit reference, since sync makes the source and the dest identical (modifying destination only).
+The check command can be used as a success/failure indication on the exported objects, since it checks if the files in the source and destination match, and logs a report of files which don't match.
+
+In order to use Rclone, users will need to download it if necessary, learn it, and have it in their toolset.
+
+Usage example:
+
+```shell
+rclone copy source:lakefs:example-repo/main/ dest:s3://example-bucket/prefix
+
+rclone sync source:lakefs:example-repo/main/ dest:s3://example-bucket/prefix
+
+rclone check source:lakefs:example-repo/main/ dest:s3://example-bucket/prefix
+```
+
+Pros:
+- Rclone gives the functionality of exporting data from lakeFS to a designated object store location. Require only few additional work of documentation for the usage of Rclone as an export functionality.
+- Simple to use. 
+- Doesn't require maintaining a new feature.
+
+Cons:
+- Require a designated object store location that doesn't contain any other data but the data associated to the lakeFS branch. That is because Rclone's sync command will delete all files that don't exist in the branch.   
+
+
+## Chosen solution
+
+Wrap Rclone with a bash script to match the behavior of the new export functionality to the one of the Spark client.
```

Comment:
in the rclone section it says the user needs to download and use rclone.
Does our solution still require that?
Can you add a note about how our solution is going to be delivered?